### PR TITLE
chore: add error handling hook (swapper)

### PR DESCRIPTION
**Pull request by elmutt**
## Description

Add `showErrorToast()` utility to translate and display known errors. Currently only handles swapper errors with translations but can easily be expanded.

Replace existing swap error handling to use showErrorToast(e)

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

Low risk. Nothing should change from a user perspective just how we handle errors under the hood

## Testing

Nothing should change from a user perspective. Test that swaps still work

## Screenshots (if applicable)

This screenshot should never happen under normal circumstances but is an example of what an error will look like to the user.

Screen Shot 2022-05-19 at 3 05 25 PM

Error logs example:
Screen Shot 2022-05-19 at 3 53 35 PM

### CODE REVIEW

File: src/assets/translations/en/main.json [hunk 1]
```diff
@@ -58,7 +58,8 @@
     "login": "Log In",
     "verify": "Verify",
     "notRobot": "I am not a Robot",
-    "skip": "Skip"
+    "skip": "Skip",
+    "generalError": "An error has occured"
```
Comment (cjthompson) [inline src/assets/translations/en/main.json:62]: There is an errors key. You should put this there.

File: src/components/Trade/TradeInput.tsx [hunk 2]
```diff
@@ -78,6 +81,19 @@ export const TradeInput = ({ history }: RouterProps) => {
     if (!wallet) return
     if (!(quote?.sellAsset && quote?.buyAsset && quote.sellAmount)) return
 
+    const minSellAmount = toBaseUnit(quote.minimum, quote.sellAsset.precision)
+
+    if (bnOrZero(quote.sellAmount).lt(minSellAmount)) {
+      toast({
```
Comment (cjthompson) [inline src/components/Trade/TradeInput.tsx:87]: in a follow-up PR we could look at making `showErrorToast` support passing in translation strings in addition to errors so we can replace this `toast` call with that.

**Review by cjthompson — APPROVED** [review]

**Comment by cjthompson** [comment]
The error chaining in that screenshot is awesome!

File: src/components/Approval/Approval.tsx [hunk 3]
```diff
@@ -49,71 +47,54 @@ export const Approval = () => {
...
 
-      history.push({ pathname: TradeRoutePaths.Confirm, state: { fiatRate } })
-    }, 5000)
-  }
+        await updateTrade({
+          wallet,
+          sellAsset: quote?.sellAsset,
+          buyAsset: quote?.buyAsset,
+          amount: quote?.sellAmount,
+        })
```
Comment (theoboldfrazier) [inline src/components/Approval/Approval.tsx:91]:
```suggestion
        await updateTrade({ wallet, ...quote })
```

**Review by theoboldfrazier — APPROVED** [review]

**Pull request merged**